cpp/examples/parquet_io/parquet_io_multithreaded.cpp

```diff
@@ -0,0 +1,335 @@
+         curr_file_idx += thread_count) {
+      auto const source_info = cudf::io::source_info(input_files[curr_file_idx]);
+      auto builder           = cudf::io::parquet_reader_options::builder(source_info);
+      auto const options     = builder.build();
+      tables_this_thread.push_back(cudf::io::read_parquet(options, stream).tbl);
+    }
+
+    // Concatenate all tables read by this thread.
+    auto table = std::move(tables_this_thread[0]);
+    std::for_each(tables_this_thread.begin() + 1, tables_this_thread.end(), [&](auto& tbl) {
+      std::vector<cudf::table_view> const table_views{table->view(), tbl->view()};
+      table = cudf::concatenate(table_views, stream);
+    });
+
+    // Done with this stream
+    stream.synchronize_no_throw();
+
+    tables[thread_id] = std::move(table);
+  }
+};
+
+struct write_fn {
+  std::string const& output_path;
+  std::vector<cudf::table_view> const& table_views;
+  cudf::io::column_encoding const encoding;
+  cudf::io::compression_type const compression;
+  std::optional<cudf::io::statistics_freq> const stats_level;
+  int const thread_id;
+  rmm::cuda_stream_view stream;
+
+  void operator()()
+  {
+    // write the data for inspection
+    auto sink_info =
+      cudf::io::sink_info(output_path + "/table_" + std::to_string(thread_id) + ".parquet");
+    auto builder = cudf::io::parquet_writer_options::builder(sink_info, table_views[thread_id])
+                     .compression(compression)
+                     .stats_level(stats_level.value_or(cudf::io::statistics_freq::STATISTICS_NONE));
+    auto table_metadata = cudf::io::table_input_metadata{table_views[thread_id]};
+
+    std::for_each(table_metadata.column_metadata.begin(),
+                  table_metadata.column_metadata.end(),
+                  [=](auto& col_meta) { col_meta.set_encoding(encoding); });
+
+    builder.metadata(table_metadata);
+    auto options = builder.build();
+
+    // Write parquet data
+    cudf::io::write_parquet(options, stream);
+
+    // Done with this stream
+    stream.synchronize_no_throw();
+  }
+};
+
+int main(int argc, char const** argv)
+{
+  std::string input_paths;
+  std::string output_path;
+  cudf::io::column_encoding encoding;
+  cudf::io::compression_type compression;
+  std::optional<cudf::io::statistics_freq> page_stats;
+  int thread_count;
+
+  switch (argc) {
+    case 1:
+      input_paths  = "example.parquet";
+      output_path  = std::filesystem::current_path().string();
+      encoding     = get_encoding_type("DELTA_BINARY_PACKED");
+      compression  = get_compression_type("ZSTD");
+      thread_count = 2;
+      break;
+    case 7: page_stats = get_page_size_stats(argv[6]); [[fallthrough]];
+    case 6:
+      input_paths  = std::string{argv[1]};
+      output_path  = std::string{argv[2]};
+      encoding     = get_encoding_type(argv[3]);
+      compression  = get_compression_type(argv[4]);
+      thread_count = std::max(thread_count, std::stoi(std::string{argv[5]}));
```

Comment:
Somehow I'm ending up with a thread_count of 94!!! 

I changed line 115 to initialize thread_count to 1 and now this behaves correctly.